glide-core/src/client/value_conversion.rs

```diff
@@ -522,14 +560,66 @@ fn convert_flat_array_to_array_of_pairs(
     Ok(Value::Array(result))
 }
 
+/// Converts a flattened array with key-values to Maps.  The inner values may be another flattened array with key-values.
+/// OR
+/// Converts a Map with inner values being a flattened array with key-values.
+/// This is useful for Streams that are Keys of Stream-Ids of Fields, where the returned structure is a Map-of-Maps-of-Maps.
+/// `value` is a flattened array of key-arrays or a map of key-arrays
+fn convert_to_xread_map(value: Value) -> RedisResult<Value> {
+    match value {
+        Value::Array(array) => {
+            if !array.is_empty() && matches!(array[0], Value::Array(_)) {
+                convert_array_to_map(
+                    array,
+                    |map, key, value| {
+                        map.push((key, convert_to_xread_map(value)?));
+                        Ok(())
+                    },
+                    |map, key, value| {
+                        map.push((key, value));
+                        Ok(())
+                    },
+                )
+            } else {
+                convert_array_elements(array, ExpectedReturnType::BulkString)
+            }
+        }
+        Value::Map(map) => {
+            let map_clone = map.clone();
+            let result = map
+                .into_iter()
+                .map(|(key, inner_value)| {
+                    let key_str = match key {
+                        Value::BulkString(_) => key,
+                        Value::SimpleString(_) => key,
+                        _ => Value::BulkString(from_owned_redis_value::<String>(key)?.into()),
+                    };
+                    match inner_value {
+                        Value::Array(_) => {
+                            Ok((key_str, convert_to_xread_map(inner_value.clone())?))
```

Comment:
Are there any performance concerns regarding cloning the map above and cloning the inner array here? Or do we expect it to be fine?